app/code/Magento/InventoryCatalogAdminUi/Ui/DataProvider/Product/Form/Modifier/SourceItems.php

```diff
@@ -132,13 +124,21 @@ public function modifyMeta(array $meta)
     {
         $product = $this->locator->getProduct();
 
-        if ($this->isSingleSourceMode->execute() === true
-            || $this->isSourceItemsAllowedForProductType->execute($product->getTypeId()) === false) {
+        if ($this->isSingleSourceMode->execute() === true) {
```

Comment:
Not sure whether we should remove the `isSourceItemsAllowedForProductType` check here. 
@maghamed / @naydav : We need further feedback about this.